.github/actions/optimized_build/action.yaml

```diff
@@ -35,6 +38,13 @@ runs:
 
         echo "${{ inputs.release_version }}" >> "./artifacts/release_version"
 
+        DANGER_SKIP_CARGO_LOCK_UPDATE='0'
```

Comment:
a couple of things:
- let's refrain from using extreme-meaning words, as here "DANGER"
- waaaaay easier to read and maintain when the naming and logic do not involve negations. Rename to `CHECK_DEPENDENCIES_UPDATED`.
- for value use words/strings rather than digits referring to semantics in some distinguished programming languages